templates/test/ci/prow-topology/base.yaml

```diff
@@ -0,0 +1,264 @@
+apiVersion: cluster.x-k8s.io/v1beta1
```

Comment:
What would you think about moving this to a new clusterclass directory directly under templates so it can also be used by users too? or are we thinking of documenting a template for the "default" azure cluster separately? 

I guess this is more a general question that isn't specific to CAPZ about how providers should handle the UX around clusterclass.

Reply:
I have updated the templates as follows:
1. Cluster class definition is named as `prow-clusterclass-ci-default`. This follows capi convention as @sbueringer had suggested so that it creates the required cluster class within the namespace.
2. Cluster using the class is named as `prow-topology`.
wdyt?